Assets/Scripts/Items/ItemObject.cs

```diff
@@ -0,0 +1,75 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+/*
+    Nadklasa dla przedmiotów.
+    Nie używać jako komponentu, to NIE jest komponent.
+    Nie dodawać mu MonoBehaviour - jest usuwany przy destroy.
+    Pojedyncza instancja jest przekazywana jako obiekt (jest z nim równoważna) w porównaniu do komponentów
+    które nie zachowują tych własności
+*/
+public abstract class ItemObject : MonoBehaviour {
+
+    protected Rigidbody rigid;
+    protected Collider collid;
+    protected MeshRenderer renderer;
+    protected float pickUpDelay = 0f;
+
+    void Start()
+    {
+        rigid = gameObject.GetComponent<Rigidbody>();
+        collid = gameObject.GetComponent<Collider>();
+        renderer = gameObject.GetComponent<MeshRenderer>();
+    }
+
+    void Update()
+    {
+        if (pickUpDelay > 0f)
+        {
+            pickUpDelay -= Time.deltaTime;
```

Comment:
This could be replaced with `pickUpDelay = Mathf.Max(0f, pickUpDelay - Time.deltaTime)` if that's all it should do